### PR TITLE
Docs process updated

### PR DESCRIPTION
**Pull request by kumare3**
- Better named makefile targets to generate docs
- Also using a pinned version of Flytekit and FlyteIDL for docs
- Added links to go repos documentation
- Fixed the index and introduction page (there was a duplicate)

### CODE REVIEW

File: script/generate_docs.sh [hunk 1]
```diff
@@ -16,13 +16,13 @@ RSTS_DIR=`mktemp -d "${BASEDIR}/rsts_tmp/XXXXXXXXX"`
 
 # Copy all rst files to the same place
 cp -R rsts/* ${RSTS_DIR}
-cp -R _rsts/* ${RSTS_DIR}
+#cp -R _rsts/* ${RSTS_DIR}
```
Comment (wild-endeavor) [inline script/generate_docs.sh:19]: I don't remember, what does this do?

**Comment by wild-endeavor** [comment]
+1 with one comment.  approving, not sure if you want to change it

File: script/generate_docs.sh (hunk 1 above)
Comment (kumare3) [reply script/generate_docs.sh]: i think we need this, good catch, added back

**Comment by kumare3** [comment]
> but the github workflow doesn't push to gh-pages automatically right? we still need to do that separately.

Right, that should be a follow up PR, I have something in progress, mind helping?

**Comment by wild-endeavor** [comment]
yes ofc.

**Pull request merged**